python/ray/_raylet.pyx

```diff
@@ -452,13 +452,9 @@ cdef execute_task(
         actor_id = core_worker.get_actor_id()
         actor = actor_class.__new__(actor_class)
         worker.actors[actor_id] = actor
-        # Record the actor name via :actor_name: magic token in the log file.
-        # This is used for the prefix in driver logs `(actor_repr pid=123)...`
-        if (hasattr(actor_class, "__ray_actor_class__") and
-                "__repr__" in actor_class.__ray_actor_class__.__dict__):
-            print("{}{}".format(
-                ray_constants.LOG_PREFIX_ACTOR_NAME, repr(actor)))
-        else:
+        if (<int>task_type == <int>TASK_TYPE_ACTOR_CREATION_TASK):
+            # Record the actor class via :actor_name: magic token in the log.
+            # (Phase 1): this covers code run before __init__ finishes.
```

Comment:
```suggestion
            # (Phase 1): this covers code run before __init__ finishes. We need to handle this
            # separately because `__repr__` may not be runnable until after `__init__` (e.g., if
            # it accesses fields initialized in the constructor).
```

Reply:
Done